starknet_py/cairo/serialization/data_serializers/named_tuple_serializer.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass
+from typing import Union, Dict, NamedTuple, OrderedDict, Generator
+
+from starknet_py.cairo.serialization._context import (
+    DeserializationContext,
+    SerializationContext,
+)
+from starknet_py.cairo.serialization.data_serializers._common import (
+    deserialize_to_dict,
+    serialize_from_dict,
+)
+from starknet_py.cairo.serialization.data_serializers.cairo_data_serializer import (
+    CairoDataSerializer,
+)
+from starknet_py.utils.tuple_dataclass import TupleDataclass
+
+
+@dataclass
+class NamedTupleSerializer(
+    CairoDataSerializer[Union[Dict, NamedTuple, TupleDataclass], TupleDataclass]
+):
+    """
+    Serializer for tuples with named fields.
+    Can serialize a dictionary, a named tuple and TupleDataclass.
+    Deserializes data to a TupleDataclass.
+    """
+
+    serializers: OrderedDict[str, CairoDataSerializer]
+
+    def deserialize_with_context(
+        self, context: DeserializationContext
+    ) -> TupleDataclass:
+        as_dictionary = deserialize_to_dict(self.serializers, context)
+        return TupleDataclass.from_dict(as_dictionary)
+
+    def serialize_with_context(
+        self,
+        context: SerializationContext,
+        value: Union[Dict, NamedTuple, TupleDataclass],
+    ) -> Generator[int, None, None]:
+        # We can't use isinstance(value, NamedTuple), because there is no NamedTuple type.
+        context.ensure_valid_type(
+            value,
+            isinstance(value, (dict, TupleDataclass)) or self._is_namedtuple(value),
```

Comment:
Shouldn't we use `Dict` here?

Reply:
No, Dict is the name for the type checking, but dict is the type (https://stackabuse.com/python-check-if-variable-is-a-dictionary/). Fun, right? :D

Reply:
Yup, very funny 🙄